From: Michael Chen <michaelchen644@gmail.com>
Date: Thu, 30 May 2025 10:00:00 +0800
Subject: [PATCH] fix(nft): update nft_fullcone_validate signature to match Linux 6.12+
Linux 6.12+ changes the validate() callback in nft_expr_ops from 3 to 2
parameters. This patch updates the fullcone NAT extension accordingly.
Signed-off-by: Michael Chen <michaelchen644@gmail.com>
diff --git a/nft_ext_fullcone_new.c b/nft_ext_fullcone.c
index 8f4ce21..96d1feb 100644
--- a/src/nft_ext_fullcone.c
+++ b/src/nft_ext_fullcone.c
@@ -121,7 +121,7 @@ static int exp_event_cb(unsigned int events, const struct nf_exp_event *item)
 }
 #endif
 
-static int nft_fullcone_validate(const struct nft_ctx *ctx, const struct nft_expr *expr, const struct nft_data **data)
+static int nft_fullcone_validate(const struct nft_ctx *ctx, const struct nft_expr *expr)
 {
 	int err;
 
